### PR TITLE
Lint

### PR DESCRIPTION
**Pull request by viniciussalvati**
Configured eslint to be used with typescript. It will also load some configs from prettier. The options are commented in the config file itself.

I choose to not enforce the rules when running `npm start` cause it would display the errors in the browser and whould not render anything until fixed, and that would be very annoying.

I also fixed the errors that appeared in the files.

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    'react-app', // Uses recommended rules from react-scripts
+  ],
+  rules: {
+    // Place to specify ESLint rules. Can be used to overwrite rules specified from the extended configs
+    // eslint rules: https://eslint.org/docs/rules/
+    // @typescript-eslint rules: https://github.com/typescript-eslint/typescript-eslint/tree/master/packages/eslint-plugin
+    '@typescript-eslint/explicit-function-return-type': 'off', // with this we won't need to specify functions return types. Typescript will infer it for us
+    'no-console': 'warn', // Lets not leave console.logs all over
+    curly: 'warn', // forces { and } after ifs, while, for, etc
+    indent: ['warn', 2],
```
Comment (viniciussalvati) [inline .eslintrc.js:32]: I prefer tab indentation instead of two spaces, but this is what I saw everywhere I looked. I'm open to opinions.

Comment (bellesamways) [reply .eslintrc.js:32]: my vote is two spaces

Comment (viniciussalvati) [reply .eslintrc.js:32]: I configured it for two spaces

**Review by bellesamways — APPROVED** [review]

**Review by pedrofracassi — APPROVED** [review]
LGTM! Can we copy that to the other repositories?

**Comment by viniciussalvati** [comment]
@pedrofracassi Yes, but it would need some changes. React scripts already come with some packages that are required, we would need to install those, as well as install and configure typescript. And also remove the rules specific to typescript, but it's totally doable.

Create an issue in the repo you want and we'll take a look at it.

**Pull request merged**